packages/typescript/src/node/typescript-contribution.ts

```diff
@@ -71,6 +71,13 @@ export class TypeScriptContribution extends BaseLanguageServerContribution {
         if (isRequestMessage(message)) {
             if (message.method === InitializeRequest.type.method) {
                 const initializeParams = message.params as TypeScriptInitializeParams;
+                const capabilities = initializeParams.capabilities;
+                if (capabilities.textDocument) {
+                    if (!capabilities.textDocument.documentSymbol) {
+                        capabilities.textDocument.documentSymbol = {};
+                    }
+                    capabilities.textDocument.documentSymbol.hierarchicalDocumentSymbolSupport = true;
```

Comment:
@akosyakov, do you know why this capability isn't `true` by default in Theia?

Reply:
it should be: https://github.com/TypeFox/vscode-languageserver-node/blob/fb80c5e2db4c714032c28d6f482e770642ae9e23/client/src/client.ts#L1574

Reply:
well, this isn't the case. I'm investigating, why `capabilities.textDocument.documentSymbol` is not present at all.

Reply:
thanks for the pointer, BTW!

Reply:
nvm, I had a type in `fillClientCapabilities`.